### PR TITLE
TP2000-143: 500 on Measure Detail view

### PR DESCRIPTION
**Pull request by gabelton**
## Why
It shouldn't be possible to create a measure with the same `sid` as another measure from a different `version_group`. There's still a separate issue being explored in this ticket ( https://uktrade.atlassian.net/browse/TP2000-42 ) around synchronising measure sids so that data engineers can make changes alongside those of Tariff Managers on the frontend, but this should add a bit of extra security in the meantime.

## What
- Adds custom validation in Measure.clean()
- Adds unit test 
- Specifies marksupsafe version in requirements.txt (https://github.com/aws/aws-sam-cli/issues/3661)

### CODE REVIEW

**Review by simonwo — COMMENTED** [review]
Is it possible to do this in a general way and apply it anywhere else we have a SID? Also, are there any implications for this being a validation and not a business rule e.g. will it be possible to import duplicate SIDs and fix them later, or will this prevent that?

**Comment by gabelton** [comment]
> 

I think a 500 error would block import, so maybe a business rule is better. Might make generalising it for other components a bit easier too

**Comment by gabelton** [comment]
I added a test for measures just because that was what originally motivated the PR, but I think tests in common/tests/test_business_rules should already have us covered.

File: measures/models.py [hunk 1]
```diff
@@ -279,6 +281,7 @@ class MeasureType(TrackedModel, ValidityMixin):
         business_rules.MT4,
         business_rules.MT7,
         business_rules.MT10,
+        UniqueIdentifyingFields,
```
Comment (simonwo) [inline measures/models.py:284]: I think MT1 already does this one.

File: measures/models.py [hunk 2]
```diff
@@ -53,6 +54,7 @@ class MeasureTypeSeries(TrackedModel, ValidityMixin):
     business_rules = (
         business_rules.MTS1,
         business_rules.MTS2,
+        UniqueIdentifyingFields,
```
Comment (simonwo) [inline measures/models.py:57]: MTS1 already does this I think.

File: commodities/models/orm.py [hunk 3]
```diff
@@ -176,6 +177,7 @@ def footnote_application_codes(self) -> Set[ApplicationCode]:
         business_rules.NIG31,
         business_rules.NIG34,
         business_rules.NIG35,
+        UniqueIdentifyingFields,
```
Comment (simonwo) [inline commodities/models/orm.py:180]: Ah I think in this case NIG1 actually has a slightly different requirement.

**Review by simonwo — COMMENTED** [review]
There are a few other models that already have rules that cover uniqueness… can you double check we're not doubling up?

**Review by simonwo — APPROVED** [review]

**Pull request merged**